.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,73 @@
+name: benchmark
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
As far as I understand it, it is only discouraged when using github pages publishing, because then anyone could push to the github pages branch.
But Mapperly for now only tracks the delta between the default branch and the feature branch via github actions cache. Therefore this should not a problem. If this job significantly slows down the CI we could add a limitation to only run it if the PR is labeled, but this can be adjusted as we use it.